packages/squiggle-lang/src/utility/compactTreeString.ts

```diff
@@ -0,0 +1,77 @@
+import { makeAndToString } from "./FormattedNumber.js";
+import { ImmutableMap } from "./immutable.js";
+
+function formatNumber(number: number): string {
+  return makeAndToString(number, { precision: 3, forceScientific: true });
+}
+
+function formatNumberArray(numbers: number[]): string[] {
+  return numbers.map(formatNumber);
+}
+
+// Converts a tree structure into a compact string representation.
+// Useful for debugging, including sending to LLMs.
+// T can be SimpleValue or SimpleValueWithoutLambda
+export function compactTreeString<T>(
+  value: T,
+  depth: number = 0,
+  maxDepth: number = 5,
+  maxArrayItems: number = 5,
+  maxDictItems: number = 30
```

Comment:
Five position parameters is too many, change to `(value: T, opts?: { ... } = {})`?